src/js/libgeo/GeoOps.js

```diff
@@ -1928,14 +2018,23 @@ geoMacros.Calculation = function(el) {
     return [];
 };
 
+geoMacros.Arc = function(el) {
+    el.type = "ArcBy3";
+    return [el];
+};
+
 geoMacros.Transform = function(el) {
-    var trKind = csgeo.csnames[el.args[0]].kind;
-    var argKind = csgeo.csnames[el.args[1]].kind;
+    var arg = csgeo.csnames[el.args[1]];
+    var tr = csgeo.csnames[el.args[0]];
+    // workaround for Arcs since we treat them as circles
+    var akind = arg.kind;
+    if (arg.kind === "C" && arg.type === "ArcBy3") akind = "Arc";
```

Comment:
Good point, i think we need  ```arg.type === "ArcBy3" || arg.isArc``` since the macros are evaluated before initialization takes place.

I thought about introducing a new type, but since cinderella never had real use cases despite ArcBy3 i abandoned the idea.